File: 0201-0300/0233-Number of Digit One/0233-Number of Digit One.cpp
class Solution {
public:
    int countDigitOne(int n) {
        int count = 0;
        for (long base = 1; base <= n; base *= 10) {
            int curr = n / base % 10;
            int low = n % base;
            int high = n / (base * 10);
            if (curr < 1) {
                count += high * base;
            }
            else if (curr == 1) {
                count += high * base + low + 1;
            }
            else {
                count += (high + 1) * base;
            }
        }
        
        return count;
    }
};

File: 0201-0300/0233-Number of Digit One/0233-Number of Digit One.py
class Solution:
    def countDigitOne(self, n: int) -> int:
        base = 1
        count = 0
        while base <= n:
            curr = n // base % 10
            low = n % base
            high = n // (base * 10)
            if curr < 1:
                count += high * base
            elif curr == 1:
                count += high * base + low + 1
            else:
                count += (high + 1) * base
            base *= 10
        return count
